support-lambdas/bigquery-acquisitions-publisher/typescript/schemas.ts

```diff
@@ -0,0 +1,363 @@
+	'SJ',
+	'SK',
+	'SM',
+	'TF',
+	'TR',
+	'WF',
+	'YT',
+	'VA',
+	'AX',
+	'KI',
+	'NR',
+	'NF',
+	'TV',
+	'AE',
+	'AF',
+	'AG',
+	'AI',
+	'AM',
+	'AO',
+	'AQ',
+	'AR',
+	'AS',
+	'AW',
+	'AZ',
+	'BB',
+	'BD',
+	'BF',
+	'BH',
+	'BI',
+	'BJ',
+	'BM',
+	'BN',
+	'BO',
+	'BQ',
+	'BR',
+	'BS',
+	'BT',
+	'BV',
+	'BW',
+	'BY',
+	'BZ',
+	'CC',
+	'CD',
+	'CF',
+	'CG',
+	'CI',
+	'CL',
+	'CM',
+	'CN',
+	'CO',
+	'CR',
+	'CU',
+	'CV',
+	'CW',
+	'CX',
+	'DJ',
+	'DM',
+	'DO',
+	'DZ',
+	'EC',
+	'EG',
+	'EH',
+	'ER',
+	'ET',
+	'FJ',
+	'FM',
+	'GA',
+	'GD',
+	'GE',
+	'GH',
+	'GM',
+	'GN',
+	'GQ',
+	'GS',
+	'GT',
+	'GU',
+	'GW',
+	'GY',
+	'HK',
+	'HM',
+	'HN',
+	'HT',
+	'ID',
+	'IL',
+	'IN',
+	'IO',
+	'IQ',
+	'IR',
+	'JM',
+	'JO',
+	'JP',
+	'KE',
+	'KG',
+	'KH',
+	'KM',
+	'KN',
+	'KP',
+	'KR',
+	'KW',
+	'KY',
+	'KZ',
+	'LA',
+	'LB',
+	'LC',
+	'LK',
+	'LR',
+	'LS',
+	'LY',
+	'MA',
+	'MD',
+	'MG',
+	'MH',
+	'MK',
+	'ML',
+	'MM',
+	'MN',
+	'MO',
+	'MP',
+	'MR',
+	'MS',
+	'MU',
+	'MV',
+	'MW',
+	'MX',
+	'MY',
+	'MZ',
+	'NA',
+	'NC',
+	'NE',
+	'NG',
+	'NI',
+	'NP',
+	'NU',
+	'OM',
+	'PA',
+	'PE',
+	'PG',
+	'PH',
+	'PK',
+	'PN',
+	'PR',
+	'PS',
+	'PW',
+	'PY',
+	'QA',
+	'RU',
+	'RW',
+	'SA',
+	'SB',
+	'SC',
+	'SD',
+	'SG',
+	'SL',
+	'SN',
+	'SO',
+	'SR',
+	'SS',
+	'ST',
+	'SV',
+	'SX',
+	'SY',
+	'SZ',
+	'TC',
+	'TD',
+	'TG',
+	'TH',
+	'TJ',
+	'TK',
+	'TL',
+	'TM',
+	'TN',
+	'TO',
+	'TT',
+	'TW',
+	'TZ',
+	'UA',
+	'UG',
+	'UM',
+	'UY',
+	'UZ',
+	'VC',
+	'VE',
+	'VG',
+	'VI',
+	'VN',
+	'VU',
+	'WS',
+	'YE',
+	'ZA',
+	'ZM',
+	'ZW',
+	'FK',
+	'GI',
+	'GG',
+	'IM',
+	'JE',
+	'SH',
+]);
+export type IsoCountry = z.infer<typeof IsoCountrySchema>;
+
+export const PaymentProviderSchema = z.enum([
+	'STRIPE',
+	'STRIPE_APPLE_PAY',
+	'STRIPE_PAYMENT_REQUEST_BUTTON',
+	'STRIPE_SEPA',
+	'PAYPAL',
+	'GOCARDLESS',
+	'IN_APP_PURCHASE',
+]);
+export type PaymentProvider = z.infer<typeof PaymentProviderSchema>;
+
+const PrintProductSchema = z.enum([
+	'HOME_DELIVERY_EVERYDAY',
+	'HOME_DELIVERY_EVERYDAY_PLUS',
+	'HOME_DELIVERY_SIXDAY',
+	'HOME_DELIVERY_SIXDAY_PLUS',
+	'HOME_DELIVERY_WEEKEND',
+	'HOME_DELIVERY_WEEKEND_PLUS',
+	'HOME_DELIVERY_SATURDAY',
+	'HOME_DELIVERY_SATURDAY_PLUS',
+	'HOME_DELIVERY_SUNDAY',
+	'HOME_DELIVERY_SUNDAY_PLUS',
+	'NATIONAL_DELIVERY_EVERYDAY',
+	'NATIONAL_DELIVERY_SIXDAY',
+	'NATIONAL_DELIVERY_WEEKEND',
+	'VOUCHER_EVERYDAY',
+	'VOUCHER_EVERYDAY_PLUS',
+	'VOUCHER_SIXDAY',
+	'VOUCHER_SIXDAY_PLUS',
+	'VOUCHER_WEEKEND',
+	'VOUCHER_WEEKEND_PLUS',
+	'VOUCHER_SATURDAY',
+	'VOUCHER_SATURDAY_PLUS',
+	'VOUCHER_SUNDAY',
+	'VOUCHER_SUNDAY_PLUS',
+	'GUARDIAN_WEEKLY',
+]);
+
+export const PrintOptionsSchema = z
+	.object({ product: PrintProductSchema, deliveryCountry: IsoCountrySchema })
+	.nullable();
+export type PrintOptions = z.infer<typeof PrintOptionsSchema>;
+
+export const ReaderTypeSchema = z.enum(['Direct', 'Gift']);
+export type ReaderType = z.infer<typeof ReaderTypeSchema>;
+
+// This defines the schema for the data we expect to receive in the acquisition event.
```

Comment:
Shall we have a comment somewhere here linking to the scala model? We now have to keep the two models in sync

Reply:
Yes that's a good shout 👍  Eventually we'd like to migrate the backend code to TS (Rupert is already working on re-writing support-workers in TS) so in future we can hopefully share the TS model everywhere.

Reply:
Added in b66edf5.